docs/how-to/write-devices.md

```diff
@@ -0,0 +1,70 @@
+# Write Devices for Blueapi
+
+:::{seealso}
+[Home of Plans and Devices](../explanations/extension-code.md) for information about where device code usually lives.
+:::
+
+
+## Format
+
+:::{seealso}
+[Dodal](https://github.com/DiamondLightSource/dodal) for many more examples
+:::
+
+Devices are made using the [dodal](https://github.com/DiamondLightSource/dodal) style available through factory functions like this:
+
+```python 
+from my_facility_devices import MyTypeOfDetector
+
+def my_detector(name: str) -> MyTypeOfDetector:
+    return MyTypeOfDetector(name, {"other_config": "foo"})
+```
+
+The return type annotation `-> MyTypeOfDetector` is required as blueapi uses it to determine that this function creates a device. Meaning you can have a Python file where only some functions create devices and they will be automatically picked up. Similarly, these functions can be organized per-preference into files. 
+
+The device is created via a function rather than a global to preserve side-effect-free imports. Each device must have its own factory function.
+
+# How to Configure Detectors to Write Files
+
+:::{note}
+**This is an absolute requirement to write data onto the Diamond Filesystem**. This decorator must be used every time a new data collection is intended to begin. For an example, see below.
+:::
+
+Dodal defines a decorator, `@attach_metadata`, for configuring `ophyd-async` detectors to write data to a common location. 
```

Comment:
must:
https://github.com/DiamondLightSource/dodal/blob/a3cf0ecf432dc7ad2a62c2583e90d452154f9ef5/src/dodal/plan_stubs/data_session.py#L43
```suggestion
Dodal defines a decorator, `@attach_data_session_metadata_decorator()`, for configuring `ophyd-async` detectors to write data to a common location. 
```

could: explain that it requires the get_path_provider from dodal is used as constructor arg?